ui/src/components/PopInput/PopInput.jsx

```diff
@@ -42,32 +42,20 @@ export default class PopInput extends React.Component {
     this.save = this.save.bind(this);
     this.cancel = this.cancel.bind(this);
     this.submit = this.submit.bind(this);
-    this.showOvelay = this.showOvelay.bind(this);
-    this.state = {show: false};
+    this.hideOverlay = this.hideOverlay.bind(this);
   }
 
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.data.state !== this.props.data.state) {
-      if (this.isIdle(nextProps.data)) {
-        this.handleIdle(nextProps.data);
-      } else if (this.isAborted(nextProps.data)) {
-        this.handleError(nextProps.data);
-      } else {
-        this.handleError(nextProps.data);
-      }
-    }
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      nextProps.data.state !== this.props.data.state ||
+      nextProps.data.value !== this.props.data.value
+    );
   }
 
-
-  showOvelay(value) {
-    this.setState({
-      show: value
-    });
+  hideOverlay() {
+    document.body.click();
```

Comment:
Letting the `OverlayTrigger`  handle the state show internally by triggering `document.body.click()`

Reply:
The React Way:tm: would have been to set a `ref` on the trigger element, but this is much simpler :joy:

Reply:
Yes I know, but I honestly find that so clonky, and the API have been changing. So when I use the easier solutions when I find one (and probably faster, its apparently a thing now a days :) )